### ISSUE TITLE
Constant Median Filter  Run time greater than 200ms

### ISSUE THREAD

**Issue by DLFCW** [Ldpe2G/ArmNeonOptimization#2]
In 3.4Ghz

**Comment by DLFCW** [comment]
thank you for your reply

**Comment by DLFCW** [comment]
> @DLFCW My cpu is 3.9Ghz, the run time of constant median filter with radius 5 and image size 1280x1024 is about 140 ms.
> And the `HISTOGRAM_LEN` should be 256 not 512, https://github.com/Ldpe2G/ArmNeonOptimization/blob/master/ConstantTimeMedianFilter/src/constant_time_median_filter_uint16.h#L8
> After you change it to 256, you should see a little speed up.
> And by the way, if your image size is too large and filter radius is small, it is not recommended to use this algorithm, because you need to allocate a large chunck of memory to store the column histograms.
> You can simply try to use the parallel strategy like the normal median filter dose:
> https://github.com/Ldpe2G/ArmNeonOptimization/blob/master/ConstantTimeMedianFilter/src/normal_median_filter_uint16.cpp#L27

do you know halcon? a machine vision library  .
The library run constant median filter only 0.9ms in same condition

**Comment by Ldpe2G** [comment]
No, have not heard before, the library must be done a lot optimization. I have just implemented the basic algorithm described in the paper, and there are some optimization tips described in the paper that  I did not try.
